src/App.tsx

```diff
@@ -15,35 +18,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  default = '',
+  length = 'length',
+  alphabetically = 'alphabetically',
+  reverse = 'reverse',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: SortType,
+  reverseField: boolean,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.length:
+          return good1.length - good2.length;
+
+        case SortType.alphabetically:
+          return good1.localeCompare(good2);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return reverseField ? preparedGoods.reverse() : preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.default);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortField, isReversed);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== SortType.alphabetically,
+          })}
+          onClick={() => {
+            setSortField(SortType.alphabetically);
+          }}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== SortType.length,
+          })}
+          onClick={() => {
+            setSortField(SortType.length);
+          }}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': isReversed !== true,
+          })}
+          onClick={() => {
+            if (isReversed === false) {
+              setIsReversed(true);
+            } else {
+              setIsReversed(false);
+            }
```

Comment:
Move this logic from jsx to the helper function and use it here
```suggestion
            if (!isReversed) {
              setIsReversed(true);
            } else {
              setIsReversed(false);
            }
```